### PR TITLE
[13.0] [MIG] stock_lot_scrap: Migration to v13

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT27135

please, @carlosdauden @joao-p-marques can you review this?

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
Code review

File: stock_lot_scrap/models/stock_production_lot.py [hunk 1]
```diff
@@ -72,7 +71,18 @@ def action_scrap_lot(self):
...
+        scrap_location_id = (
+            self.env["stock.location"]
+            .search(
+                [
+                    ("company_id", "=", self.company_id.id),
+                    ("scrap_location", "=", True,),
+                ]
+            )
+            .id
+        )
```
Comment (joao-p-marques) [inline stock_lot_scrap/models/stock_production_lot.py]: I think this search may cause problems in the future, you can make it more resilient by limiting the search results to 1:
```suggestion
        # The data model has disappeared and right now is a function who creates the
        # scrap location. That's why we have to search by company_id and scrap_location.
        scrap_location_id = (
            self.env["stock.location"]
            .search(
                [
                    ("company_id", "=", self.company_id.id),
                    ("scrap_location", "=", True,),
                ],
                limit=1,
            )
            .id
        )

File: stock_lot_scrap/models/stock_production_lot.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    _inherit = "stock.production.lot"
+
+    @api.model
+    def fields_view_get(
+        self, view_id=None, view_type="form", toolbar=False, submenu=False
+    ):  # pragma: no cover
+        """Inject the button here to avoid conflicts with other modules
+         that add a header element in the main view.
+        """
+        res = super(StockProductionLot, self).fields_view_get(
```
Comment (joao-p-marques) [inline stock_lot_scrap/models/stock_production_lot.py]:
```suggestion
        res = super().fields_view_get(
```

File: stock_lot_scrap/models/stock_scrap.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+class StockScrap(models.Model):
+    _inherit = "stock.scrap"
+
+    def action_validate(self):
+        self.ensure_one()
+        self.lot_id.message_post(
+            body=_("Lot was scrapped by <b>%s</b>.") % self.env.user.name
+        )
+        return super(StockScrap, self).action_validate()
```
Comment (joao-p-marques) [inline stock_lot_scrap/models/stock_scrap.py]:
```suggestion
        return super().action_validate()
```

**Comment by CarlosRoca13** [comment]
@joao-p-marques Changes done!

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "restructuredtext.confPath": ""
+}
```
Comment (joao-p-marques) [inline .vscode/settings.json]: It seems like this was added here by mistake. It shouldn't go here.

Comment (CarlosRoca13) [reply .vscode/settings.json]: Oh true, sorry. Removed

**Review by joao-p-marques — APPROVED** [review]
Code review :+1:

File: .vscode/settings.json (hunk 4 above)
Comment (joao-p-marques) [reply .vscode/settings.json]: No problem. I always got the same issue :sweat_smile:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**